src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/UseNameofInPlaceOfString.cs

```diff
@@ -0,0 +1,141 @@
+
+    {
+        // TODO: RuleId 
+        internal const string RuleId = "NAMEOFANALYZER";
+        private const string ParamName = "paramName";
+        private const string PropertyName = "propertyName";
+
+        // TODO: need final wording for feature
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftMaintainabilityAnalyzersResources.UseNameOfInPlaceOfStringTitle), MicrosoftMaintainabilityAnalyzersResources.ResourceManager, typeof(MicrosoftMaintainabilityAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftMaintainabilityAnalyzersResources.UseNameOfInPlaceOfStringMessage), MicrosoftMaintainabilityAnalyzersResources.ResourceManager, typeof(MicrosoftMaintainabilityAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftMaintainabilityAnalyzersResources.UseNameOfInPlaceOfStringDescription), MicrosoftMaintainabilityAnalyzersResources.ResourceManager, typeof(MicrosoftMaintainabilityAnalyzersResources));
+
+        internal static DiagnosticDescriptor RuleWithSuggestion = new DiagnosticDescriptor(RuleId,
+                                                                         s_localizableTitle,
+                                                                         s_localizableMessage,
+                                                                         DiagnosticCategory.Maintainability,
+                                                                         DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                         isEnabledByDefault: true,
+                                                                         description: "Use nameof",
+                                                                         // TODO: add MSDN url
+                                                                         helpLinkUri: "http://msdn.microsoft.com/library/ms182181.aspx",
+                                                                         customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleWithSuggestion);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterOperationAction(AnalyzeArgument, OperationKind.Argument);
+        }
+
+        private void AnalyzeArgument(OperationAnalysisContext context)
+        {
+
+            var argument = (IArgumentOperation)context.Operation;
+
+            if (argument.Value.Kind != OperationKind.Literal)
```

Comment:
Also check if `argument.Value.Type.SpecialType` is [SpecialType.System_String](http://source.roslyn.io/#Microsoft.CodeAnalysis/SpecialType.cs,126), otherwise bail out.